packages/core/src/testE2E/amazonQTransform/transformByQ.test.ts

```diff
@@ -0,0 +1,86 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as path from 'path'
+import { setValidConnection } from '../util/connection'
+import assert from 'assert'
+import { JDKVersion, TransformationType, transformByQState } from '../../codewhisperer'
+import {
+    processLanguageUpgradeTransformFormInput,
+    setMaven,
+    startTransformByQ,
+} from '../../codewhisperer/commands/startTransformByQ'
+import { fs } from '../../shared'
+import { TestFolder } from '../../test/testUtil'
+
+describe('transformByQ', async function () {
+    let tempDir = ''
+    let tempFileName = ''
+    let tempFilePath = ''
+    let validConnection: boolean
+
+    const javaFileContents = `public class MyApp {
+            public static void main(String[] args) {
+                Integer temp = new Integer("1234");
+            }
+        }`
+
+    const pomXmlContents = `<?xml version="1.0" encoding="UTF-8"?>
+        <project xmlns="http://maven.apache.org/POM/4.0.0"
+                xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+            <modelVersion>4.0.0</modelVersion>
+
+            <groupId>com.example</groupId>
+            <artifactId>basic-java-app</artifactId>
+            <version>1.0-SNAPSHOT</version>
+
+            <properties>
+                <maven.compiler.source>1.8</maven.compiler.source>
+                <maven.compiler.target>1.8</maven.compiler.target>
+                <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+            </properties>
+
+            <build>
+                <plugins>
+                    <plugin>
+                        <groupId>org.apache.maven.plugins</groupId>
+                        <artifactId>maven-compiler-plugin</artifactId>
+                        <version>3.8.1</version>
+                        <configuration>
+                            <source>1.8</source>
+                            <target>1.8</target>
+                        </configuration>
+                    </plugin>
+                </plugins>
+            </build>
+        </project>`
+
+    before(async function () {
+        validConnection = await setValidConnection()
+        if (!validConnection) {
+            this.skip()
+        }
+        tempDir = path.join((await TestFolder.create()).path, 'qct-java-upgrade-test')
+        tempFileName = 'MyApp.java'
+        tempFilePath = path.join(tempDir, tempFileName)
+        await fs.writeFile(tempFilePath, javaFileContents)
+        tempFileName = 'pom.xml'
+        tempFilePath = path.join(tempDir, tempFileName)
+        await fs.writeFile(tempFilePath, pomXmlContents)
+    })
+
+    // TODO: this test 1) is skipped in GitHub CI due to no valid connection (see line 62 above) and
```

Comment:
Previously I remember this being true. But 1) I don't think this test is being invoked in CI on this PR? and 2) I see: https://github.com/aws/aws-toolkit-vscode/blob/38850f7c941cc4fbd56e9b9bcb8bd5681c7dbf73/packages/amazonq/test/e2e/amazonq/featureDev.test.ts#L116-L118 — guessing that is necessary to run E2E tests that hit our backend endpoint?

Reply:
@jpinkney-aws may know ^